meilisearch-http/src/helpers/meilisearch.rs

```diff
@@ -225,26 +166,24 @@ impl<'a> SearchBuilder<'a> {
 
         let start = Instant::now();
         let result = query_builder.query(reader, &self.query, self.offset..(self.offset + self.limit));
-        let (docs, nb_hits) = result.map_err(|e| Error::SearchDocuments(e.to_string()))?;
+        let (docs, nb_hits) = result.map_err(ResponseError::search_documents)?;
         let time_ms = start.elapsed().as_millis() as usize;
 
         let mut hits = Vec::with_capacity(self.limit);
         for doc in docs {
             // retrieve the content of document in kv store
-            let mut fields: Option<HashSet<&str>> = None;
-            if let Some(attributes_to_retrieve) = &self.attributes_to_retrieve {
-                let mut set = HashSet::new();
-                for field in attributes_to_retrieve {
-                    set.insert(field.as_str());
-                }
-                fields = Some(set);
-            }
+            let attributes: Option<HashSet<&str>> = self
+                .attributes_to_retrieve
+                .as_ref()
+                .map(|a| a.iter().map(|a| a.as_str()).collect());
 
             let document: IndexMap<String, Value> = self
                 .index
-                .document(reader, fields.as_ref(), doc.id)
-                .map_err(|e| Error::RetrieveDocument(doc.id.0, e.to_string()))?
-                .ok_or(Error::DocumentNotFound(doc.id.0))?;
+                .document(reader, attributes.as_ref(), doc.id)
+                .map_err(|e| ResponseError::retrieve_document(doc.id.0, e))?
+                .ok_or(ResponseError::internal(
+                    "Impossible to retrieve a document id returned by the engine",
```

Comment:
I'm not sure to understand the error message here. This error occurs when there was no document for that id?

Reply:
This error is an internal error. It will be triggered if we have data corruption. The search method told me there is a document in the database but it's impossible to retrieve it.